### ISSUE TITLE
Time out error when downloading images

### ISSUE THREAD

**Issue by fragpic** [eclipse-che/che#16545]
### Describe the bug
Times out when scheduling/downloading images.
Logs indicate timeout waiting for postgres volumes to be bound
 

### Che version

 - [x] latest

### Steps to reproduce
ran 
chectl server:start --platform minikube --multiuser

### Expected behavior
Eclipse che is deployed and a url is generated

### Runtime
   - [ ] kubernetes (include output of `kubectl version`)
   - [ ] Openshift (include output of `oc version`)
   - [x] minikube (include output of `minikube version` and `kubectl version`)
   - [ ] minishift (include output of `minishift version` and `oc version`)
   - [ ] docker-desktop + K8S (include output of `docker version` and `kubectl version`)
   - [ ] other: (please specify)

minikube version: v1.9.1
commit: d8747aec7ebf8332ddae276d5f8fb42d3152b5a1

Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:50:46Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}

### Screenshots
  ✔ Verify Kubernetes API...OK
  ✔ 👀  Looking for an already existing Eclipse Che instance
    ✔ Verify if Eclipse Che is deployed into namespace "che"...it is not
  ✔ ✈️  Minikube preflight checklist
    ✔ Verify if kubectl is installed
    ✔ Verify if minikube is installed
    ✔ Verify if minikube is running
    ↓ Start minikube [skipped]
      → Minikube is already running.
    ✔ Check Kubernetes version: Found v1.18.0.
    ✔ Verify if minikube ingress addon is enabled
    ✔ Enable minikube ingress addon
    ✔ Retrieving minikube IP and domain for ingress URLs...172.17.0.2.nip.io.
Eclipse Che logs will be available in '/tmp/chectl-logs/1586030573786'
  ✔ Start following logs
    ✔ Start following Operator logs...done
    ✔ Start following Eclipse Che logs...done
    ✔ Start following Postgres logs...done
    ✔ Start following Keycloak logs...done
    ✔ Start following Plugin registry logs...done
    ✔ Start following Devfile registry logs...done
  ✔ Start following events
    ✔ Start following namespace events...done
  ✔ 🏃‍  Running the Eclipse Che operator
    ✔ Copying operator resources...done.
    ✔ Create Namespace (che)...It already exists.
    ✔ Create ServiceAccount che-operator in namespace che...It already exists.
    ✔ Create Role che-operator in namespace che...It already exists.
    ✔ Create ClusterRole che-operator...It already exists.
    ✔ Create RoleBinding che-operator in namespace che...It already exists.
    ✔ Create ClusterRoleBinding che-operator...It already exists.
    ✔ Create CRD checlusters.org.eclipse.che...It already exists.
    ✔ Waiting 5 seconds for the new Kubernetes resources to get flushed...done.
    ✔ Create deployment che-operator in namespace che...It already exists.
    ✔ Create Eclipse Che cluster eclipse-che in namespace che...It already exists.
  ❯ ✅  Post installation checklist
    ❯ Eclipse Che pod bootstrap
      ✖ scheduling
        → ERR_TIMEOUT: Timeout set to pod wait timeout 300000. podExist: false, currentPhase: undefined
        downloading images
        starting
      Retrieving Eclipse Che server URL
      Eclipse Che status check
 ›   Error: Error: ERR_TIMEOUT: Timeout set to pod wait timeout 300000. podExist: false, currentPhase: undefined
 ›   Installation failed, check logs in '/tmp/chectl-logs/1586030573786'

### Installation method
   - [x] chectl <-->chectl server:start --platform minikube --multiuser

### Environment
   - [x ] my computer
       - [ ] Windows
       - [x] Linux
       - [ ] macOS
   - [ ] Cloud
       - [ ] Amazon
       - [ ] Azure
       - [ ] GCE
       - [ ] other (please specify)
   - [ ] other: please specify

### Eclipse Che Logs
time="2020-04-04T20:02:45Z" level=info msg="Default 'info' log level is applied"
time="2020-04-04T20:02:45Z" level=info msg="Go Version: go1.12.12"
time="2020-04-04T20:02:45Z" level=info msg="Go OS/Arch: linux/amd64"
time="2020-04-04T20:02:45Z" level=info msg="operator-sdk Version: v0.5.0"
time="2020-04-04T20:02:45Z" level=info msg="Operator is running on Kubernetes"
time="2020-04-04T20:02:45Z" level=info msg="Registering Che Components Types"
time="2020-04-04T20:02:45Z" level=info msg="Starting the Cmd"
time="2020-04-04T20:02:45Z" level=info msg="Waiting for PVC postgres-data to be bound. Default timeout: 10 seconds"
time="2020-04-04T20:02:55Z" level=warning msg="Timeout waiting for a PVC postgres-data to be bound. Current phase is Pending"
time="2020-04-04T20:02:55Z" level=warning msg="Sometimes PVC can be bound only when the first consumer is created"
time="2020-04-04T20:02:56Z" level=info msg="Waiting for deployment postgres. Default timeout: 420 seconds"

**LAST SEEN**|**TYPE**|**REASON**|**OBJECT**|**MESSAGE**
:-----:|:-----:|:-----:|:-----:|:-----:
22m|Normal|Scheduled|pod/che-operator-7b9fd956cb-fwbt8|Successfully assigned che/che-operator-7b9fd956cb-fwbt8 to minikube
22m|Normal|Pulling|pod/che-operator-7b9fd956cb-fwbt8|Pulling image "quay.io/eclipse/che-operator:7.10.0"
21m|Normal|Pulled|pod/che-operator-7b9fd956cb-fwbt8|Successfully pulled image "quay.io/eclipse/che-operator:7.10.0"
21m|Normal|Created|pod/che-operator-7b9fd956cb-fwbt8|Created container che-operator
21m|Normal|Started|pod/che-operator-7b9fd956cb-fwbt8|Started container che-operator
18s|Normal|SandboxChanged|pod/che-operator-7b9fd956cb-fwbt8|Pod sandbox changed, it will be killed and re-created.
16s|Normal|Pulling|pod/che-operator-7b9fd956cb-fwbt8|Pulling image "quay.io/eclipse/che-operator:7.10.0"
13s|Normal|Pulled|pod/che-operator-7b9fd956cb-fwbt8|Successfully pulled image "quay.io/eclipse/che-operator:7.10.0"
13s|Normal|Created|pod/che-operator-7b9fd956cb-fwbt8|Created container che-operator
13s|Normal|Started|pod/che-operator-7b9fd956cb-fwbt8|Started container che-operator
22m|Normal|SuccessfulCreate|replicaset/che-operator-7b9fd956cb|Created pod: che-operator-7b9fd956cb-fwbt8
22m|Normal|ScalingReplicaSet|deployment/che-operator|Scaled up replica set che-operator-7b9fd956cb to 1
6m46s|Warning|FailedScheduling|pod/postgres-6448d66f7f-2hn8w|running "VolumeBinding" filter plugin for pod "postgres-6448d66f7f-2hn8w"$
9s|Warning|FailedScheduling|pod/postgres-6448d66f7f-2hn8w|running "VolumeBinding" filter plugin for pod "postgres-6448d66f7f-2hn8w"$
21m|Normal|SuccessfulCreate|replicaset/postgres-6448d66f7f|Created pod: postgres-6448d66f7f-2hn8w
6m30s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
21m|Normal|ScalingReplicaSet|deployment/postgres|Scaled up replica set postgres-6448d66f7f to 1
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Warning|FailedScheduling|pod/postgres-6448d66f7f-2hn8w|running "VolumeBinding" filter plugin for pod "postgres-6448d66f7f-2hn8w"$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Warning|FailedScheduling|pod/postgres-6448d66f7f-2hn8w|running "VolumeBinding" filter plugin for pod "postgres-6448d66f7f-2hn8w"$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner "k8s.i$
0s|Normal|ExternalProvisioning|persistentvolumeclaim/postgres-data|waiting for a volume to be created, either by external provisioner

**Comment by tolusha** [comment]
@cbyreddy 
That's might be the cause. https://github.com/kubernetes/minikube/issues/7218
pls. downgrade minikube to v1.8 and try again.

**Comment by gattytto** [comment]
@tolusha the storage provisioner error applies in postgres pod creation stage

the error in che-server startup stage applies for domain name resolution errors normally (either inside che-server or in the cluster layer).

option a: edit /usr/local/lib/chectl/templates/che-operator/crds/org_v1_che_cr.yaml
specifically the line:
````yaml
ingressDomain: 'xxxxxxxxx.nip.io' 
````
then make sure to add `--che-operator-cr-yaml=/usr/local/lib/chectl/templates/che-operator/crds/org_v1_che_cr.yaml ` to the chectl start command 

option b: cluster-side dns name resolution error
this can happen when minikube starts the cluster without specifying resolve.conf location and the file in /etc/resolv.conf is being managed by systemd-resolved. The solution for this is to stop and disable systemd-resolved systemd unit and manually remove the symlink (etc/resolv.conf) and create the file again with static "nameserver x.x.x.x" lines.

**Comment by gattytto** [comment]
https://github.com/che-incubator/chectl/pull/461

**Comment by cbyreddy** [comment]
> @cbyreddy
> That's might be the cause. [kubernetes/minikube#7218](https://github.com/kubernetes/minikube/issues/7218)
> pls. downgrade minikube to v1.8 and try again.

I was having issues with v1.8 too so I tried again using k3s. I got to a much later stage of the install process before it errored out again.


    sudo chectl server:start --platform k8s  --multiuser --domain 192.168.1.137.nip.io
      ✔ Verify Kubernetes API...OK
      ✔ 👀  Looking for an already existing Eclipse Che instance
        ✔ Verify if Eclipse Che is deployed into namespace "che"...it is not
      ✔ ✈️  Kubernetes preflight checklist
        ✔ Verify if kubectl is installed
        ✔ Verify remote kubernetes status...done.
        ✔ Check Kubernetes version: Found v1.17.4+k3s1.
        ✔ Verify domain is set...set to 192.168.1.137.nip.io.
        ✔ Check if cluster accessible... ok
    Eclipse Che logs will be available in '/tmp/chectl-logs/1586681575409'
      ✔ Start following logs
        ↓ Start following Operator logs [skipped]
        ✔ Start following Eclipse Che logs...done
        ✔ Start following Postgres logs...done
        ✔ Start following Keycloak logs...done
        ✔ Start following Plugin registry logs...done
        ✔ Start following Devfile registry logs...done
      ✔ Start following events
        ✔ Start following namespace events...done
       ✔ 🏃‍  Running Helm to install Eclipse Che
        ✔ Verify if helm is installed
        ✔ Check Helm Version: Found v3.1.2+gd878d4d
        ✔ Create Namespace (che)...done.
        ✔ Check Eclipse Che TLS certificate...going to generate self-signed one
          ✔ Check Cert Manager deployment...not deployed
          ✔ Deploy cert-manager...done
          ✔ Wait for cert-manager...ready
          ✔ Check Cert Manager CA certificate...generating new one
          ✔ Set up Eclipse Che certificates issuer...done
          ✔ Request self-signed certificate...done
          ✔ Wait for self-signed certificate...ready
          ✔ ❗[MANUAL ACTION REQUIRED] Please add local Eclipse Che CA certificate into your browser: /home/admin/cheCA.crt
        ✔ Check Cluster Role Binding...does not exists.
        ✔ Preparing Eclipse Che Helm Chart...done.
        ✔ Updating Helm Chart dependencies...done.
        ✔ Deploying Eclipse Che Helm Chart...done.
      ❯ ✅  Post installation checklist
        ✔ PostgreSQL pod bootstrap
          ✔ scheduling...done.
          ✔ downloading images...done.
          ✔ starting...done.
        ✔ Devfile registry pod bootstrap
          ✔ scheduling...done.
          ✔ downloading images...done.
          ✔ starting...done.
        ✔ Plugin registry pod bootstrap
          ✔ scheduling...done.
          ✔ downloading images...done.
          ✔ starting...done.
        ❯ Eclipse Che pod bootstrap
          ✔ scheduling...done.
          ✔ downloading images...done.
          ✖ starting
            → ERR_TIMEOUT: Timeout set to pod ready timeout 130000
          Retrieving Eclipse Che server URL
          Eclipse Che status check
        Show important messages
     ›   Error: Error: ERR_TIMEOUT: Timeout set to pod ready timeout 130000
     ›   Installation failed, check logs in '/tmp/chectl-logs/1586681575409'

Any idea what could have gone wrong now? I don't think it is a storage provisioning issue but I'm not sure. 
There are so many logs in the chectl folder. Which one of them would have the information pertaining to this specific error.

**Comment by zarinfam** [comment]
I have had this problem with both `microk8s` and `k3s`!

**Comment by tolusha** [comment]
@cbyreddy 
`chectl version` ?

Pls provide logs for the second installation.
https://pastebin.com/9NatW21j isn't available

**Comment by cbyreddy** [comment]
> @cbyreddy
> `chectl version` ?
> 
> Pls provide logs for the second installation.
> https://pastebin.com/9NatW21j isn't available

Sorry, here you go.
https://pastebin.com/aTY9zLRn

This seems to be the issue but I'm not sure

`Caused by: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: No name matching keycloak-che.192.168.1.137.nip.io found`

**Comment by tolusha** [comment]
related one
https://github.com/eclipse/che/issues/16429

**Comment by cbyreddy** [comment]
> related one
> #16429

Is there anything I can try to fix the error or is it a bug?

**Comment by tolusha** [comment]
It is a bug. I will take a look on it later.

What I can suggest for now:
- use a lower version of minikube (I personally use 1.5.2)
- deploy che with helm installer:
```
chectl server:start --platform minikube --installer helm --multiuser
```

**Comment by tolusha** [comment]
@cbyreddy 
Could you specify the chectl version you used?

**Comment by tolusha** [comment]
@zarinfam 
Could you specify the chectl version you used?

**Comment by tolusha** [comment]
@cbyreddy 
I am closing this one. 
Feel free to open a new issue.

**Comment by cwchoe** [comment]
I have a same problem on Azure AKS.. 

command: chectl server:start --installer=helm --platform=k8s --domain=azr.skcc-ide.cloud --multiuser

---Eclipse Che logs will be available in '/tmp/chectl-logs/1593133798522'
  ✔ Start following logs
    ↓ Start following Operator logs [skipped]
    ✔ Start following Eclipse Che logs...done
    ✔ Start following Postgres logs...done
    ✔ Start following Keycloak logs...done
    ✔ Start following Plugin registry logs...done
    ✔ Start following Devfile registry logs...done
  ✔ Start following events
    ✔ Start following namespace events...done
 ›   Warning: Consider using the more reliable 'OLM' installer when deploying a stable release of Eclipse Che (--installer=olm).
  ✔ 🏃‍  Running Helm to install Eclipse Che
    ✔ Verify if helm is installed
    ✔ Check Helm Version: Found v3.2.4+g0ad800e
    ✔ Create Namespace (che)...does already exist.
    ✔ Check Eclipse Che TLS certificate...TLS certificate secret found
    ✔ Check Cluster Role Binding...does not exists.
    ✔ Preparing Eclipse Che Helm Chart...done.
    ✔ Updating Helm Chart dependencies...done.
    ✔ Deploying Eclipse Che Helm Chart...done.
  ❯ ✅  Post installation checklist
    ✔ PostgreSQL pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ✔ Devfile registry pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ✔ Plugin registry pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ❯ Eclipse Che pod bootstrap
      ✔ scheduling...done.
      ✖ downloading images
        → ERR_TIMEOUT: Timeout set to pod wait timeout 300000
        starting
      Retrieving Eclipse Che server URL
      Eclipse Che status check
    Show important messages
 ›   Error: Error: ERR_TIMEOUT: Timeout set to pod wait timeout 300000
 ›   Installation failed, check logs in '/tmp/chectl-logs/1593133798522'

Do you have any idea?
